dice_ml/explainer_interfaces/explainer_base.py

```diff
@@ -139,6 +142,8 @@ def _generate_counterfactuals(self, query_instance, total_CFs,
         pass
 
     def setup(self, features_to_vary, permitted_range, query_instance, feature_weights):
+        self.data_interface.continuous_feature_indexes = [query_instance.columns.get_loc(name) for name in
+                                                          self.data_interface.continuous_feature_names]
         if features_to_vary == 'all':
```

Comment:
Same as above comment.

Reply:
> Looks good to me now. @gaugup Can you confirm? Now the code allows outcome to be any position in the columns, not necessarily last.

The fix looks fine. I have a comment where this property is being set. Ideally this should be implemented in public/private data interface.

Reply:
The reason we can't do that is because the continuous feature indices should be set according to the order of columns in the query_instance, and we get this information only during `generate_counterfactuals`

Reply:
But we can certainly move the logic to the data interface and call the function from here